blockstore/apps/api/v1/views/drafts.py

```diff
@@ -0,0 +1,179 @@
+
+        2. Bundles can become very large, and a PUT might become prohibitively
+        large. Having everything as a PATCH lets us set somewhat sane per-PATCH
+        request limits and let the client handle the case where we need to do
+        multiple requests to make the necessary changes.
+
+        3. It's just simpler to have only one way to update the files.
+        """
+        serializer = DraftFileUpdateSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        files_to_write = serializer.validated_data
+
+        draft_repo = DraftRepo(SnapshotRepo())
+        draft_repo.update_files(uuid, files_to_write)
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @action(detail=True, methods=['post'])
+    def commit(self, request, uuid):
+        """
+        Commit the Draft and create a new BundleVersion that points to it.
+
+        In the future, we may want to separate these two steps so that we can
+        create multiple BundleVersions at once in a single transaction, however
+        given our modeling conventions of a course being in a single Bundle,
+        that's not something that we need to implement immediately.
+
+        We currently return a summary of the things that were created, however
+        we may need to rethink this interface if the commit process is going to
+        take so long as to require async processing.
+
+        TODO: Test with large Bundles.
+        """
+        draft_repo = DraftRepo(SnapshotRepo())
+        staged_draft = draft_repo.get(uuid)
+
+        # Is this the appropriate response when trying to commit a Draft with
+        # no changes?
+        if not staged_draft.files_to_overwrite:
+            raise serializers.ValidationError("Draft has no changes to commit.")
+
+        new_snapshot, _updated_draft = draft_repo.commit(staged_draft)
+        new_bv = BundleVersion.create_new_version(
+            new_snapshot.bundle_uuid, new_snapshot.hash_digest
+        )
+
+        # This is a placeholder response. May need to revisit after trying
+        # some large commits.
+        result = {
+            'bundle_version': reverse(
+                'api:v1:bundleversion-detail',
+                args=[new_snapshot.bundle_uuid, new_bv.version_num],
+                request=request,
+            ),
+            'snapshot': reverse(
+                'api:v1:snapshots-detail',
+                args=[new_snapshot.bundle_uuid, new_snapshot.hash_digest.hex()],
+                request=request,
+            ),
+            'updated_draft': reverse(
+                'api:v1:draft-detail',
+                args=[uuid],
+                request=request,
+            )
+        }
+        return Response(result, status=status.HTTP_201_CREATED)
+
+    def destroy(self, request, uuid):
+        """
+        This removes any files that were staged along with the database entry.
+        """
+        draft_repo = DraftRepo(SnapshotRepo())
```

Comment:
Nit: `draft_repo = DraftRepo(SnapshotRepo())` comes up a lot; is there a reason not to make it a long-lived/global singleton or module-level var? Or make a property access to get it as `self.draft_repo` ?